### ISSUE TITLE
Policy Loss with Minimum or Q1?

### ISSUE THREAD

**Issue by pranv** [pranz24/pytorch-soft-actor-critic#3]
In line:
https://github.com/pranz24/pytorch-soft-actor-critic/blob/master/sac.py#L125

should it not it be this?
`policy_loss = ((self.alpha * log_prob) - q1_new).mean()`

**Comment by pranv** [comment]
Sorry, I think your implementation is correct given that the paper says:
> We then use the minimum of the Q-functions for the value gradient in Equation 6 and policy gradient in Equation 1

I was misled by OpenAI spinning up's implementation [here](https://github.com/openai/spinningup/blob/master/spinup/algos/sac/sac.py#L176). I guess I'll open an issue there

**Comment by pranv** [comment]
Just to conclude: actually, authors themselves use q1, but they [say](https://github.com/haarnoja/sac/issues/14) it does not make much difference.

**Comment by pranz24** [comment]
Thank you, for the thorough evaluation, anyway.

**Comment by wayunderfoot** [comment]
But in TD3impletion,the author use Q1 for the policy update.
